### ISSUE TITLE
Legality of cards that are not legal nor banned in any format

### ISSUE THREAD

**Issue by MatiasOETamminen** [PokemonTCG/pokemon-tcg-data#247]
I'm logging this issue based on conversation in the Discord server.
The legalities field cannot be left empty, so at least one legality has to be marked. However, there seems to be only two values: "Legal" and "Banned". There is a number of cards that are neither in any of the three formats (Unlimited/Expanded/Standard), so correctly displaying the legality of those cards is not currently possible if we understand "Banned" as being singled out (by TPCi on their ban lists) as non-legal out of otherwise-legal pool of cards.

The affected cards are the cards with the rule "(This card cannot be used at official tournaments.)":

- Imakuni?'s Doduo (xy12-112)
- Dragapult (swshp-SWSH132)
- Zacian LV.X (swshp-SWSH135)
- Mimikyu δ (swshp-SWSH136)
- Light Toxtricity (swshp-SWSH137)
- Hydreigon C (swshp-SWSH138)
- Greninja Star (swshp-SWSH144)

I think that either a third state of legality should be added for cards that are inherently not part of the pool of legal cards ("Not legal"?) or we should agree to mark these cards as Unlimited: "Banned" even though they aren't in the sense mentioned above.

**Comment by CptSpaceToaster** [comment]
`"Not legal"` is a great suggestion! I would accept it in a heartbeat. 

Have you considered `"Unplayable"`? It might make sense to suggest that there's an issue with "the card interfering with how the game is played fundamentally"

Looking at the other examples you listed, it seems like "Unplayable" is a bad suggestion, as a number of them are just promos that exist, but aren't _legal yet_

**Comment by Dillonzer** [comment]
Legality should just be a `boolean` value. It's either playable or not. If you really want to go into details I suggest adding a `legality-reasoning` property that you can fill in with a string.

**Comment by adback03** [comment]
If I simply made it possible to have an empty legality block, would this resolve these issues and be more clear to users?

**Comment by MatiasOETamminen** [comment]
> If I simply made it possible to have an empty legality block, would this resolve these issues and be more clear to users?
> 
> Otherwise, perhaps a third state makes most sense and just always include legality info for all formats for all cards.

I'd be happy with either, so I vouch for whichever is easier to implement.

**Comment by MatiasOETamminen** [comment]
> Looking at the other examples you listed, it seems like "Unplayable" is a bad suggestion, as a number of them are just promos that exist, but aren't _legal yet_

This isn't exactly the case. Those promos are the ones made in the style of card types past. They have the same disclaimer than Imakuni?'s Doduo, "(This card cannot be used at official tournaments.)", so they won't become legal.

**Comment by CptSpaceToaster** [comment]
I think the best result from what I'm reading here, is to convert the existing behavior into a boolean... introducing some breaking changes to make legality just a simple boolean... so maybe something for a v3 sort of release. In the meantime, the current enumerated values "Banned" (false) and "Legal" (true) can sort of stand in their place, as long as we agree that the wording isn't as clear as it was first intended.

**Comment by adback03** [comment]
I'm not sure a boolean is the best option, only for the case that people want to know specifically if a card is banned versus simply not legal. In that case, the boolean wouldn't suffice, and we would need additional fields to convey that information.

**Comment by CptSpaceToaster** [comment]
I think the issue was trying to clear up that the cards weren't explicitly banned... just not permitted yet, or disallowed in tournament play through other means (clauses on cards, etc). Labeling them as. Unless there's meaning to be obtained as to "how the card is banned"... (which I'm not so sure needs to be made) then a boolean should suffice.

If we like the extra information i.e. Which type of "Unplayable" is this, then "Banned" versus "Not legal" versus "Legal" seems appropriate.

**Comment by jparanada** [comment]
Let me summarize for my own understanding. The current API can represent three legality states:
* legal in the format, e.g. `"standard": "Legal"`
* banned in the format, e.g. `"standard": "Banned"`
* doesn't have a printing in the format, represented by not having a key-value pair for that format at all.

We want to represent a fourth state:
* never intended for tournament play/casual-only

While we're here there is also a fifth state that no one has mentioned:
* was once legal, but has since been superseded by another card with the same name but a different effect. The older card *doesn't* get errata to be played as the newer card--instead, you simply aren't allowed to play the older card at all. e.g. [Base Set Bill](https://pokemontcg.guru/card/bill-base/base1-91) isn't usable in Unlimited, due to the existence of HeartGold & SoulSilver Bill (which was printed as a Supporter)... @immewnity can perhaps confirm my understanding of legality for this case. (Yes, that means that the Unlimited legality markings for some older cards are incorrect at the moment.)

This fifth state is probably out-of-scope for the current issue, but it's worth thinking about if you're shooting for a more heavyweight overhaul of legality.

Back to the fourth "casual-only" state. Here are some ways it could be represented:
1. Add a new string value that represents "casual-only." This doesn't break backwards compatibility so long as the "Legal" and "Banned" values are left alone. Continue to represent "not in format" by omitting the key-value pair for a given format.
2. Change the format value from string to boolean. This seems undesirable since it both breaks backwards compatibility and isn't as expressive as some of the other options.
3. Change the shape of the format value from string to object. This breaks backwards compatibility but lets us be pretty expressive as far as why a card isn't legal.
4. remove the legality key-value pairs for all formats, *and* allow "legalities" to either be`{}` or omitted altogether. This essentially combines "casual-only" with "not-in-format." Note that this is what Scryfall does for cards that are casual-only, e.g. [cards from "Un-" sets](https://scryfall.com/card/ust/108/earl-of-squirrel) and [commemorative cards](https://scryfall.com/card/hho/14/mishras-toy-workshop) (although this doesn't mean we should do the same :) ). 

I vote for either option 1 or 4 since those seem easiest to implement (and @adback03 suggested both of these in an earlier comment).

**Comment by MatiasOETamminen** [comment]
> While we're here there is also a fifth state that no one has mentioned:
> 
> was once legal, but has since been superseded by another card with the same name but a different effect. The older card _doesn't_ get errata to be played as the newer card--instead, you simply aren't allowed to play the older card at all. e.g. [Base Set Bill](https://pokemontcg.guru/card/bill-base/base1-91) isn't usable in Unlimited, due to the existence of [HeartGold & SoulSilver Bill](https://pokemontcg.guru/card/bill-heartgold-and-soulsilver/hgss1-89) (which was printed as a Supporter)... @immewnity can perhaps confirm my understanding of legality for this case. (Yes, that means that the Unlimited legality markings for some older cards are incorrect at the moment.)

This is incorrect. Functionally different reprints function differently in unlimited than in standard/expanded. I've explained it in detail here: https://github.com/PokemonTCG/pokemon-tcg-data/issues/130#issuecomment-910158059. The gist of it is that Base Set Bill is _legal_ but plays as the HGSS Bill. See Section 5.4.2.2 (page 33) of the Tournament Rules Handbook: https://assets.pokemon.com//assets/cms2/pdf/play-pokemon/rules/play-pokemon-tournament-rules-handbook-09302021-en.pdf

My understanding is that we have the following legality states:
1. Legal
2. rotated out or otherwise too old to be legal
3. explicitly banned
4. not meant for tournament play

A simple boolean would be good enough for me, but if we want to get more granular, we want to separate 3 from 2 and 4.

**Comment by immewnity** [comment]
I think we're mixing two different definitions of "legal". One defines legality regardless of format, the other depends on the format.

How's about...

```json
"card_legality": {
    "tournament_legal": true,
    },
    "standard": {
        "playable": false,
        "reason": "Rotated"
    },
    "expanded": {
        "playable": false,
        "reason": "Banned"
    }
    "unlimited": {
        "playable": true
    }
}
```

With `reason` having options for both false (`Banned`, `Rotated`) and true (`Errata`, `Reprint`). Could even have a true value like `Explicit` for cards that are normally in-format (e.g. all cards in Battle Styles at the moment, for example), though a different term would probably be clearer.
